CHANGELOG.rst

```diff
@@ -27,17 +27,23 @@ New features and notable changes:
 - Add :option:`--jacoco` to generate JaCoCo XML format. (:issue:`823`))
 - Add function coverage to HTML report. (:issue:`828`)
 - Improve sorting of data in reports (:issue:`817`):
+
   - Sort file names alpha numerical and with casefold (see `str.casefold <https://docs.python.org/3.11/library/stdtypes.html?highlight=str%20casefold#str.casefold>`_) (``file_10.c`` comes after ``file_0.c``).
   - Always sort at the end by filename if line or branch coverage is identical for a file.
   - Add :option:`--sort-branches` to sort by branches instead of lines, this is the default if :option:`--txt-branches` is used.
   - Add :option:`--sort-reverse` to reverse the sort order.
+
 - Add option to report covered lines in txt report. (:issue:`836`)
 - Add support for specifying files for :option:`search_paths`. (:issue:`834`)
 - Use different color for partial covered lines in HTML report. (:issue:`839`)
 - Add support to generate LCOV info files. (:issue:`830`)
 - Add support for FIPS enabled OS when used with Python 3.9. (:issue:`850`)
 - Reduce file size for detailed HTML reports by using CSS to link the function lists. (:issue:`840`)
 - Ignore all negative hits if :option:`--gcov-ignore-parse-errors` is used. (:issue:`852`)
+- Use literal options for sorting and TXT metric. (:issue:`867`)
+
+  - The :option:`-b`, :option:`--txt-branches` and :option:`--branches` are deprecated, use :option:`--txt-metric` instead.
+  - The :option:`--sort-uncovered` and :option:`--sort-percentage` are deprecated, use :option:`--sort` instead.
```

Comment:
I don't remember if we have a policy on what we mean by "deprecation". Perhaps it would be better to explicitly state our intention here.

* is it "will be removed in gcovr x.0.0"? I think this would cause unnecessary breakage, especially since these options are very old and will be part of some ancient scripts.
* or is it "hey FYI, we have something better now" and we are just *discouraging* use of the old options?

Reply:
`Discouraged option which can be removed in newer versions´ is maybe better. I'm not sure when to drop the support for such option.
After this PR I want to extend our logger to support the pipeline tasks:
- [Azure](https://github.com/MicrosoftDocs/azure-devops-docs/blob/main/docs/pipelines/scripts/logging-commands.md#logissue-log-an-error-or-warning)
- [GitHub](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-warning-message)

If a pipeline run is detected we can modify the format to get the errors listed.

Reply:
Deprecated is also used in the documentation and is handled in `bump_version.py`. In this case I'll stick to this. I think we should keep it for the next few versions/year.